File: 0169-majority-element/solution.java
class Solution {
    public int majorityElement(int[] nums) {
        int s=0;
        if(nums.length==1){
            return nums[0];
        }
        for(int i=0;i<nums.length;i++){
            int count=0;
            for(int j=1+i;j<nums.length;j++){
                if(nums[i]==nums[j]){
                    count++;
                }
            }
            if(count==Math.floor(nums.length/2)){
                 s=i;
            }
        }
        return nums[s];
    }
}

File: 0169-majority-element/solution.cpp
class Solution {
public:
//Boyer-Moore Voting Algorithm
    int majorityElement(vector<int>& nums) {
        int c=0;
        int e=0;
        for(int i=0;i<nums.size();i++){
            if(c==0){
                c=1;
                e=nums[i];
            }
            else if(e==nums[i])c++;
            else{
                c--;
            }
        }
        return e;
    }
};
